### PR TITLE
Removed kv_cache from HPU graph output

### PR DESCRIPTION
**Pull request by jkaniecki**
KV cache is not necessary as a graph output (and is a huge memory-consuming tensor at the same time), so it can be deleted from outputs by replacing model forward. KV cache still needs to be returned during prefill phase, where it's created, so there is a need for some conditional code.

### CODE REVIEW

File: server/text_generation_server/models/causal_lm.py [hunk 1]
```diff
@@ -496,6 +519,8 @@ def __init__(
                 ds_inference_kwargs["checkpoint"] = checkpoints_json.name
             model = deepspeed.init_inference(model, **ds_inference_kwargs)
             model = model.module
+            model = wrap_in_hpu_graph(remove_kv_cache_from_output(model), disable_tensor_cache=True)
```
Comment (kdamaszk) [inline server/text_generation_server/models/causal_lm.py]: Please add "if self.enable_hpu_graph" to be consistent

**Review by kdamaszk — APPROVED** [review]

**Comment by jkaniecki** [comment]
@regisss please review

**Review by regisss — APPROVED** [review]
LGTM!

**Comment by regisss** [comment]
@jkaniecki There is a merge conflict to solve before merging

**Comment by jkaniecki** [comment]
@regisss - resolved

**Pull request merged**